src/happy_eyeballs.ml

```diff
@@ -207,7 +237,13 @@ let connect_ip t now ~id dsts =
     | [] -> failwith "addresses are empty"
   in
   let state = Connecting (now, dst, dsts) in
-  let conn = { created = now ; ports = [] ; state ; resolved = `both } in
+  let conn = {
+    created = now ;
+    ports = [] ;
+    state ;
+    resolved = `both ;
+    resolve_left = t.resolve_retries
```

Comment:
Thinking aloud: Having `resolve_left` being positive in `connect_ip` seems a bit odd since we don't want to attempt DNS resolution of `host.invalid`. However, since `resolved` is `` `both`` we will not attempt resolving the already resolved "domain".

Reply:
indeed, fixed